web_editor/js/main.js

```diff
@@ -0,0 +1,147 @@
+// TODO The hash changes, this is very brittle. See
+// https://github.com/thedodd/trunk/issues/230 or stop using trunk.
+import { dummyData } from "./dummy_data.js";
+import init, {
+  js_way_to_lanes,
+  js_lanes_to_tags,
+} from "./osm2lanes-npm/osm2lanes_npm.js";
+await init();
+
+// Setup the toolbox controls
+new Sortable(document.getElementById("toolbox"), {
+  group: {
+    name: "toolbox",
+    pull: "clone",
+  },
+  sort: false,
+  animation: 150,
+  ghostClass: "card-being-dragged",
+});
+
+// TODO Disable the ghostClass of lanes when going here
+new Sortable(document.getElementById("delete"), {
+  group: "lanes",
+  onAdd: function (evt) {
+    var el = evt.item;
+    el.parentNode.removeChild(el);
+  },
+});
+
+// This state is set up by start_editing
+var current_road = null;
+var current_locale = null;
+
+async function start_editing() {
+  const way = BigInt(document.getElementById("osm_way_id").value);
+  // Faster dev workflow: if the way ID is the default, use baked-in data instead of waiting on Overpass.
+  var road_wrapper, locale;
+  if (way == 804788513) {
+    [road_wrapper, locale] = dummyData();
+  } else {
+    console.log(`Fetching ${way}...`);
+    [road_wrapper, locale] = await js_way_to_lanes(way);
+  }
+  const road = road_wrapper["Ok"]["road"];
+  console.log(`Got osm2lanes output, creating cards`);
+
+  current_road = JSON.parse(JSON.stringify(road));
+  current_locale = JSON.parse(JSON.stringify(locale));
+
+  // TODO Fully clean up the old cards, including whatever sortable thing is attached there?
+  const cards = document.getElementById("cards");
+  cards.replaceChildren();
+
+  // Create a card per lane
+  for (const lane of road["lanes"]) {
+    if (lane["type"] == "separator") {
```

Comment:
I'll probably hash this out in the next PR, but I might make non-draggable cards for all of the separators. It's at least important to show the center line / where directions change. Then when you drag any card across that line, we should probably flip the direction automatically.

Reply:
Closer to the second -- they're a concept in the `osm2lanes` API representing some kind of semantic or physical separator between lanes:
![Screenshot from 2022-07-27 12-12-38](https://user-images.githubusercontent.com/1664407/181233653-0f440d4c-f5e8-4fb4-b740-e0578df9f231.png)
In order to keep the osm2lanes JSON state in sync with the DOM (managed by sortable), I may insert invisible or not-movable cards to represent the separators. Not sure yet.